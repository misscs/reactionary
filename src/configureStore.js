import { applyMiddleware, createStore, compose } from 'redux'
import { composeWithDevTools } from 'redux-devtools-extension/logOnlyInProduction'
import createSagaMiddleware from 'redux-saga'

import rootReducer from 'routes/Root/rootReducer'
import rootSaga from 'routes/Root/rootSaga'

const sagaMiddleware = createSagaMiddleware()

const middlewares = [
  sagaMiddleware
]

const enhancers = [
  composeWithDevTools(applyMiddleware(...middlewares))
]

const composedEnhancer = compose(...enhancers)

/**
 * Create the store with middleware
 * @param { object } initialState - initial state
 * @return { object } store - store
 * @todo Check environment and only execute debugEnhancer if in development
 */
export default function configureStore (initialState = {}) {
  const store = createStore(
    rootReducer,
    initialState,
    composedEnhancer
  )

  // Kick off the root saga
  sagaMiddleware.run(rootSaga)

  return store
}
