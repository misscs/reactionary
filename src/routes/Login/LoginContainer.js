/*
 * Login Page
 *
 * This is the first thing users see of our App, at the '/' route
 */

import React from 'react'
import { connect } from 'react-redux'

import {
  changeUsername,
  loginRequest
} from 'routes/Login/LoginRedux'

import LoginForm from './LoginForm'
import Input from 'components/Input'

let createHandlers = function(dispatch) {
  let handleClick = function(evt) {
    if (evt !== undefined && evt.preventDefault) evt.preventDefault()
    dispatch(loginRequest())
  }

  let onChangeUsername = function(evt) {
    dispatch(changeUsername(evt.target.value))
  }

  return {
    handleClick,
    onChangeUsername
  }
}


class LoginContainer extends React.Component {
  constructor(props) {
    super(props)
    this.handlers = createHandlers(this.props.dispatch)
  }

  render() {
    return (
      <section>
        <LoginForm>
          <legend>Browse Github User Repos</legend>
          <label htmlFor="username">
            Username
          </label>
          <Input
            id="username"
            type="text"
            placeholder="Github username"
            value={this.props.username}
            onChange={this.handlers.onChangeUsername}
          />
        <button onClick={this.handlers.handleClick}>Octomagic</button>
        </LoginForm>
      </section>
    )
  }
}

export default connect()(LoginContainer)
