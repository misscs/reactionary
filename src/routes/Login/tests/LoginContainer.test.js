import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import LoginContainer from '../LoginContainer'
import configureStore from "configureStore"

describe('<LoginContainer />', () =>{
  it(`should render`, () => {
    const store = configureStore()

    const component = renderer.create(
      <Provider store={store}>
        <LoginContainer />
      </Provider>
    )

    const tree=component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
