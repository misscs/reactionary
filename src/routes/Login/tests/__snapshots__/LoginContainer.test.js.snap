// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginContainer /> should render 1`] = `
<section>
  <form
    className="gtjRNO"
  >
    <legend>
      Browse Github User Repos
    </legend>
    <label
      htmlFor="username"
    >
      Username
    </label>
    <input
      className="dALZPB"
      id="username"
      onChange={[Function]}
      placeholder="Github username"
      type="text"
      value={undefined}
    />
    <button
      onClick={[Function]}
    >
      Octomagic
    </button>
  </form>
</section>
`;
